controllers/controllers/clusters/vsphere.go

```diff
@@ -167,25 +186,138 @@ func (v *VSphereClusterReconciler) Reconcile(ctx context.Context, cluster *anywh
+		key := client.ObjectKey{
+			Namespace: capiCluster.Namespace,
+			Name:      capiCluster.Name,
+		}
+		remoteClient, err := v.tracker.GetClient(ctx, key)
+		if err != nil {
+			return reconciler.Result{}, err
+		}
+
+		v.Log.Info("About to apply CNI")
+
+		// TODO use NewCilium
+		cilium := cilium.Cilium{}
+		if err != nil {
+			return reconciler.Result{}, err
+		}
+
+		ciliumSpec, err := cilium.GenerateManifest(specWithBundles)
+		if err != nil {
+			return reconciler.Result{}, err
+		}
+		if err := reconciler.ReconcileYaml(ctx, remoteClient, ciliumSpec); err != nil {
+			return reconciler.Result{}, err
+		}
+		conditions.MarkTrue(cluster, cniSpecAppliedCondition)
+	}
+	return reconciler.Result{}, nil
+}
+
+func (v *VSphereClusterReconciler) reconcileExtraObjects(ctx context.Context, cluster *anywherev1.Cluster, capiCluster *clusterv1.Cluster, specWithBundles *c.Spec) (reconciler.Result, error) {
+	if !conditions.IsTrue(capiCluster, extraObjectsSpecPlaneAppliedCondition) {
+		extraObjects := c.BuildExtraObjects(specWithBundles)
+		if len(extraObjects) > 0 {
+			resourcesSpec := templater.AppendYamlResources(extraObjects.Values()...)
+			if err := reconciler.ReconcileYaml(ctx, v.Client, resourcesSpec); err != nil {
+				return reconciler.Result{}, err
+			}
+			conditions.MarkTrue(cluster, extraObjectsSpecPlaneAppliedCondition)
+		}
+	}
+	return reconciler.Result{}, nil
+}
+
+func (v *VSphereClusterReconciler) getCAPICluster(ctx context.Context, cluster *anywherev1.Cluster) (*clusterv1.Cluster, reconciler.Result, error) {
+	capiCluster := &clusterv1.Cluster{}
+	capiClusterName := types.NamespacedName{Namespace: "eksa-system", Name: cluster.Name}
+	v.Log.Info("Searching for CAPI cluster", "name", cluster.Name)
+	if err := v.Client.Get(ctx, capiClusterName, capiCluster); err != nil {
+		return nil, reconciler.Result{Result: &ctrl.Result{
+			Requeue:      true,
+			RequeueAfter: defaultRequeueTime,
+		}}, err
+	}
+	return capiCluster, reconciler.Result{}, nil
+}
+
+func (v *VSphereClusterReconciler) reconcileWorkerNodeSpec(ctx context.Context, cluster *anywherev1.Cluster, templateBuilder providers.TemplateBuilder, specWithBundles *c.Spec) (reconciler.Result, error) {
+	if !conditions.IsTrue(cluster, workerNodeSpecPlaneAppliedCondition) {
+		workersSpec, err := templateBuilder.GenerateCAPISpecWorkers(specWithBundles, nil, nil)
+		if err != nil {
+			return reconciler.Result{}, err
+		}
+
+		if err := reconciler.ReconcileYaml(ctx, v.Client, workersSpec); err != nil {
+			return reconciler.Result{}, err
+		}
+
+		conditions.MarkTrue(cluster, workerNodeSpecPlaneAppliedCondition)
+	}
+	return reconciler.Result{}, nil
+}
+
+func (v *VSphereClusterReconciler) reconcileControlPlaneSpec(ctx context.Context, cluster *anywherev1.Cluster, templateBuilder providers.TemplateBuilder, specWithBundles *c.Spec, cpOpt func(values map[string]interface{})) (reconciler.Result, error) {
+	if !conditions.IsTrue(cluster, controlSpecPlaneAppliedCondition) {
```

Comment:
This is not a blocker, but this will be problematic for upgrades
Also, I could be mistaken, but I thought we decided we weren't going to try to "cache" this status and we were always going to reapply the objects. Did you find any issues when doing that? Same thing applies to the conditions for the other "phases"

Reply:
yes, the focus of this PR is just the create flow. Upgrade will be done on a separate ticket